### ISSUE TITLE
Server Crash

### ISSUE THREAD

**Issue by Trnrr** [AppliedEnergistics/Applied-Energistics-2#4210]
**Describe the bug**

The server crashes with this crash log https://gist.github.com/Trnrr/099c2c15daaa9c9c620b291cacf541bb

**To Reproduce**

Im not sure, trying to find out.

**Additional context**

Crash log: https://gist.github.com/Trnrr/099c2c15daaa9c9c620b291cacf541bb
What shows in server before crash: https://gist.github.com/Trnrr/c92578d945993acf654f0e425d05f002

Possibly same issue? https://github.com/AppliedEnergistics/Applied-Energistics-2/issues/4200

**Environment**

- Minecraft Version: 1.12.2
- AE2 Version: appliedenergistics2-rv6-stable
- Forge Version: forge-1.12.2-14.23.5.2846-universal
- Spongeforge: 1.12.2-2838-7.1.7

**Comment by sagesse-cn** [comment]
No, your problem is different from #4200, it seems to be caused by a multi-threading problem.

So, does it always crash 100%? or is it because of some step?

**Comment by Trnrr** [comment]
It has been crashing starting yesterday at random times. It doesn't crash every time I load the server, it just crashes randomly. But when ever I see that error it crashes 100%. Im trying to figure out what the cause is right now and am going to test some things tomorrow.

**Comment by mindforger** [comment]
in general i would say that AE2 does not support sponge and i can see people jumping off their seats for saying this, but they seem to handle the ticking or chunk loading in a way that is not default to how vanilla does

you can try to configure your sponge to not do that but as far as i have understood the config, i could only find "async-lighting" "chunk-gc-tick-interval" that may be adjustable

if you can confim this happens without sponge, this issue can be dealt with i guess, but in the past there were only heated discussions that never ended well

**Comment by pie-flavor** [comment]
@mindforger that is flatly false. Sponge does not do ticking or chunk loading off thread. Please don't spread misinformation like that; mod developers believing things like that is why discussions get heated in the first place.

**Comment by mindforger** [comment]
sheeesh i knew it .... 

@pie-flavor  that is flatly false. Discussions don't get heated because i spread misinformation, this is based on experience and statistics in which this error happens ALMOST everytime when spong is present ... if you just couldd repeat this problem without sponge i won't say anything

they get heated because people like you jump on it like a fish to the bait

now stop bullshitting me and accept that sponge must do SOMETHING wrong ... and in the past is WAS async chunkloading (one of probably many reasons that can cause this, even oother mods can cause this)

so stop complaining and let us get to the CORE of the problem by going step by sdtep and BEGIN with testing without sponge okay?!

**Comment by pie-flavor** [comment]
> and in the past is WAS async chunkloading (one of probably many reasons that can cause this, even oother mods can cause this)

No. No it wasn't. Again, flatly false. In some cases mods (specifically IC2) were chunkloading during async lighting, but Sponge has never done async chunkloading on its own.

> so stop complaining and let us get to the CORE of the problem by going step by sdtep and BEGIN with testing without sponge okay?!

I never said Sponge wasn't the problem. I never even said it *probably* wasn't the problem. I especially never said that testing without Sponge is unnecessary. All I said was please stop saying things that are blatantly untrue about Sponge and contribute heavily to divisiveness.

**Comment by Ommina** [comment]
All of which is still a distraction from the issue at hand.

Is AE2 code at fault?  Possibly.  Is Sponge at fault?  Possibly.  Is it a THIRD mod that is doing something silly, but is throwing inside AE2?  I'd be willing to believe that too.

So, @Trnrr -- if you can do some testing without Sponge, that would be valuable.

Additionally, it would be useful to test without AE2 add-ons, on the grounds that these are the mods most likely to be interacting with the energy grid.

Again, this isn't a slight against Sponge nor any addons.  I'm confident that the respective authors will want to know.  Testing without them is nothing more than an effort to move the issue forward.

**Comment by Trnrr** [comment]
@Ommina I am currently testing our live sever with sponge async lighting disabled. I also threw up our test server and removed all ae2 addons I have. I'm not sure if that will do anything but if that doesn't I can remove the addons from the live server but I would like that to be a last ditch effort since it could possible cause world problems.

I also have a small update planned for the pack to include some mod updates. I will make sure those ae2 addons are updated and see if it still crashes after that. If it does I can remove the addons from the live server and see if it stops the crashes.

**Comment by Trnrr** [comment]
Just to give a small update:
The test server hasn't crashed. That doesn't really show anything since nobody is on it, I just have it running.

The live server hasn't crashed either since disabling async lighting. BUT I remembered I also removed the Sampler mod I had installed (https://www.curseforge.com/minecraft/mc-mods/sampler). So tomorrow I'm going to test enabling async lighting and see if it crashes. If it doesn't crash with that enabled then I will also add the sampler back in and see what happens then
